modules/programs/nix-your-shell.nix

```
@@ -0,0 +1,21 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) genAttrs getExe;
+  cfg = config.programs.nix-your-shell;
+
+  # In principle `bash` is supported too, but...  😹
+  shells = [ "fish" "ion" "nushell" "zsh" ];
+  programs = [ "nix" "nix-shell" ];
+in {
+  meta.maintainers = with lib.maintainers; [ nicoo ];
+
+  options.programs.nix-your-shell.enable = lib.mkEnableOption ''
+    `nix-your-shell`, a wrapper for `nix develop` or `nix-shell`
+    to run the same shell inside the new environment.
+  '';
+
+  config.programs = lib.mkIf cfg.enable (genAttrs shells (shell: {
+    shellAliases = genAttrs programs
+      (program: "${getExe pkgs.nix-your-shell} ${shell} ${program} --");
+  }));
+}
```

I think it may be worth simplifying the code a bit, the `genAttrs` are always a bit tricky to follow:

```suggestion
{ config, lib, pkgs, ... }:
let
  inherit (lib) getExe;
  cfg = config.programs.nix-your-shell;

  mkShellAliases = shell:
    let exe = getExe pkgs.nix-your-shell;
    in {
      nix = "${exe} ${shell} nix --";
      nix-shell = "${exe} ${shell} nix-shell --";
    };
in {
  meta.maintainers = with lib.maintainers; [ nicoo ];

  options.programs.nix-your-shell.enable = lib.mkEnableOption ''
    {command}`nix-your-shell`, a wrapper for {command}`nix develop` or
    {command}`nix-shell` to run the same shell inside the new environment'';

  config = lib.mkIf cfg.enable {
    programs.fish.shellAliases = mkShellAliases "fish";
    programs.ion.shellAliases = mkShellAliases "ion";
    programs.nushell.shellAliases = mkShellAliases "nushell";
    programs.zsh.shellAliases = mkShellAliases "zsh";
  };
}
```